src/masternodes/balances.h

```diff
@@ -254,6 +254,108 @@ struct CTransferDomainMessage {
     }
 };
 
+struct CStatsBalances {
+    TAmounts balances;
+
+    Res Add(CTokenAmount amount) {
+        if (amount.nValue == 0) {
+            return Res::Ok();
+        }
+
+        balances[amount.nTokenId] += amount.nValue;
+
+        return Res::Ok();
+    }
+
+    Res Sub(CTokenAmount amount) {
+        if (amount.nValue == 0) {
+            return Res::Ok();
+        }
+
+        balances[amount.nTokenId] -= amount.nValue;
```

Comment:
Should add safety checks on amount.nValue to ensure we are not deducting negative values.

Reply:
We will have negative here. This is intentional

Reply:
For stats, we just check out of bounds / overflow, underflow, rest is all allowed.